### DISCUSSION TITLE
How can I convert this v3 expression into v4?

### DISCUSSION THREAD

**Discussion by eudyj** [mikefarah/yq#2015] [Q&A]
I have been trying for a while (without success) to convert a script that uses yq v3 into v4. The use case here is that we're using yq to do a simple find/replace on values in a kubernetes manifest. For example, given this input:

```
apiVersion: apps/v1
kind: Deployment
metadata:
  name: _APP_NAME_
```

If my app name is "myapp", then I'd want this output:

```
apiVersion: apps/v1
kind: Deployment
metadata:
  name: myapp
```

The literal `_APP_NAME_` may appear as the value of any node in the document, any number of levels deep.

For yq v3, this replaced all occurrences in the document:

`yq write --inplace --doc "*" kubernetes-manifest.yaml '**.(.==_APP_NAME_)' myapp`

The closest I have gotten with v4 is this, but it produces extra undesirable output, and can't overwrite the file in-place

`yq '.. | select(. == "_APP_NAME_") = "myapp"' kubernetes-manifest.yaml > new_file.yaml`

The output of this is

```
apiVersion: apps/v1
kind: Deployment
metadata:
  name: myapp
apps/v1
Deployment
name: myapp
myapp
```

So, it does the replacement as desired, but also writes out every value it found afterwards.

How can I suppress all the extra output, and is there a way to have it overwrite the file in place?

**Comment by mikefarah** [comment]
You need to have brackets:

```
yq -i '(.. | select(. == "_APP_NAME_")) = "myapp"' kubernetes-manifest.yaml
```

This is to ensure you pass all the matching nodes to the `=` operator on the LHS. Otherwise, what you are doing is `..` to match all nodes, and then filtering/assigning. See https://mikefarah.gitbook.io/yq/operators/assign-update#update-deeply-selected-results for more details.

To write the file in place, pass in `-i`
